### PR TITLE
Add authenticated GeoIP database downloads

### LINKED ISSUE #128: GeoIP needs account to download
Starting December 30, 2019, downloads will no longer be served from our public GeoLite2 page, from geolite.maxmind.com/download/geoip/database/*, or from any other public URL.

https://blog.maxmind.com/2019/12/18/significant-changes-to-accessing-and-using-geolite2-databases/

--2020-03-25 13:50:35--  https://geolite.maxmind.com/download/geoip/database/GeoLite2-Country.tar.gz
Resolving geolite.maxmind.com (geolite.maxmind.com)... failed: Name or service not known.
wget: unable to resolve host address ‘geolite.maxmind.com’
--2020-03-25 13:50:35--  https://geolite.maxmind.com/download/geoip/database/GeoLite2-City.tar.gz
Resolving geolite.maxmind.com (geolite.maxmind.com)... failed: Name or service not known.
wget: unable to resolve host address ‘geolite.maxmind.com’
tar: GeoLite2-City.tar.gz: Cannot open: No such file or directory
tar: Error is not recoverable: exiting now
tar: GeoLite2-Country.tar.gz: Cannot open: No such file or directory
tar: Error is not recoverable: exiting now
mkdir: cannot create directory ‘/opt/geoip’: File exists
/home/kallai/nginx-autoinstall.sh: line 240: cd: GeoLite2-City_*/: No such file or directory
mv: cannot stat 'GeoLite2-City.mmdb': No such file or directory
/home/kallai/nginx-autoinstall.sh: line 243: cd: GeoLite2-Country_*/: No such file or directory
mv: cannot stat 'GeoLite2-Country.mmdb': No such file or directory

### PR DESCRIPTION
**Pull request by SoftCreatR**
By providing a license key from the [Maxmind licenses page](https://www.maxmind.com/en/accounts/current/license-key) (requires [registration](https://www.maxmind.com/en/geolite2/signup)), their GeoLite2 databases will be downloaded and installed. A warning will be displayed, if you attempt to install the GeoIP module without any key.

Fixes #128

### CODE REVIEW

**Comment by randomshell** [comment]
So no fallback for users without an account? @angristan @SoftCreatR

**Comment by SoftCreatR** [comment]
Well, people will find a way. I didn't check it, but I guess, a key can be obtained for free or people will find a reliable alternative by themselves. I don't think, that implementing an unofficial or unreliable fallback is a good idea.

**Comment by randomshell** [comment]
I was thinking at mirroring it here with a github action for automatic updates. See https://github.com/P3TERX/GeoLite.mmdb/blob/master/.github/workflows/GeoLite.yml

**Comment by SoftCreatR** [comment]
That's up to @angristan 

But for now, this PR is already an improvement, compared to the master code and there's still space for more improvements.

File: nginx-autoinstall.sh [hunk 1]
```diff
@@ -103,6 +103,10 @@ case $OPTION in
 		done
 		while [[ $GEOIP != "y" && $GEOIP != "n" ]]; do
 			read -rp "       GeoIP [y/n]: " -e GEOIP
+
+			while [[ $GEOIP = "y" && ! -n "$GEOIP_LICENSE_KEY" ]]; do
```
Comment (skid9000) [inline nginx-autoinstall.sh:107]: Maybe add a statement like "If no key is provided, databases will not be downloaded automatically, you will need to download them yourself and put them in /opt/geoip, etc..."

File: nginx-autoinstall.sh [hunk 2]
```diff
@@ -164,9 +168,14 @@ case $OPTION in
 			;;
 		esac
 	fi
+	if [[ $GEOIP == 'y' && ! -n "$GEOIP_LICENSE_KEY" ]]; then
+		echo ""
+		echo -e "\\033[33mAttention:\\nGEOIP_LICENSE_KEY not provided, not downloading any database for the GeoIP module!\\033[0m"
```
Comment (skid9000) [inline nginx-autoinstall.sh:173]: Attention --> Warning

**Comment by angristan** [comment]
Closed by https://github.com/angristan/nginx-autoinstall/pull/234

**Pull request closed without merging**